java/src/main/java/com/logicalclocks/hsfs/OnDemandFeatureGroup.java

```diff
@@ -108,7 +110,7 @@ public OnDemandFeatureGroup(FeatureStore featureStore, int id) {
   }
 
   public void save() throws FeatureStoreException, IOException {
-    onDemandFeatureGroupEngine.saveFeatureGroup(this);
+    onDemandFeatureGroupEngine.saveFeatureGroup(this, primaryKeys);
```

Comment:
You don't need to provide the `primaryKeys` here, they are already set in the object.